macGist/Keychain.swift

```diff
@@ -84,15 +123,124 @@ struct Keychain {
         return password
     }
     
-    func deleteItem() throws {
-        // Delete the existing item from the keychain.
-        let query = keychainQuery(withService: KeychainConfiguration.serviceName, account: KeychainConfiguration.account, accessGroup: KeychainConfiguration.accessGroup)
+    func deleteToken() throws {
+        let query = KeychainConfiguration.query()
+        let status = SecItemDelete(query as CFDictionary)
+        
+        guard status == noErr || status == errSecItemNotFound else { throw KeychainError.unhandledError(status: status) }
+    }
+    
+    func deleteGitHubClientId() throws {
+        let query = KeychainConfiguration.GitHub.Gist.Account.clientId.query()
+        let status = SecItemDelete(query as CFDictionary)
+        
+        guard status == noErr || status == errSecItemNotFound else { throw KeychainError.unhandledError(status: status) }
+    }
+    
+    func deleteGitHubClientSecret() throws {
+        let query = KeychainConfiguration.GitHub.Gist.Account.clientSecret.query()
         let status = SecItemDelete(query as CFDictionary)
         
-        // Throw an error if an unexpected status was returned.
         guard status == noErr || status == errSecItemNotFound else { throw KeychainError.unhandledError(status: status) }
     }
     
+    func save(gitHubClientId: String) throws {
+        let encodedGitHubClientId = gitHubClientId.data(using: String.Encoding.utf8)!
+        
+        do {
+            try _ = readGitHubClientId()
```

Comment:
```swift
try readGitHubClientId()
```
it isn't better?

Reply:
Adjusted and added `@discardableResult` to suppress the warning.